### DISCUSSION TITLE
Setup/Pico Functionality/Noob questions

### DISCUSSION THREAD

**Discussion by victorcapet** [markqvist/Reticulum#153] [Help & Questions] [answered]
Hello, very new to getting into the guts of programming and hardware, but this project is really interesting to me and one of my buddies and I are trying to just start by setting up a LoRa network between his house and mine. I've been reading a ton of the discussions, guides, and documentation, but I still have a few questions/things I needed confirmation on:

So as I understand it, this is how it could work and what we're aiming for: My phone running sideband > rpi gateway with wireless > attached LoRa rnode > Lora rnode (as many as needed to cover the distance) > his LoRa > rpi > phone.
- the LoRa "jumpers" in between his house and mine don't need connected rpi's, do they?
- I know there was a brief discussion in June on whether one could run a gateway on an rpi Pico: at that time @greygoo was working on an rpi/bpi zero and @markqvist was optimistic about a Pico. Any update/confirmation since then? Other rpi's are very difficult to get my hands on, as usual, but I could get pico w's very easily.

Thank you in advance from somebody who is very eager to learn and see what I can do with this really awesome project.

**Comment by markqvist** [comment, marked as answer]
Just to clear out any confusion, you can of course use *any* kind of computer as a gateway. It does not have to be a Raspbelly Pi in any way.

When those parts of the documentation were written, a Pi was still very easy and cheap to get, so it was an obvious choice for most people. Times have changed, unfortunately. So yeah, use whatever you have/can. The Reticulum stack will run on almost any kind of system, as long as you can install `python3` and `pip` (used to easily install the `rns` package).

Currently, an RNode **cannot** function as an intermediary transport node **on it's own**. You will need some kind of computer (Pi or otherwise) connected that runs `rnsd` with the transport option enabled. Even an old Android device can theoretically work out (and can in fact be used in practice, since you can enable Reticulum Transport inside the Sideband app).

In the not-too-distant future, there will be an RNS implementation that runs well on micro-controllers, like ESP32-devices and low-powered ARM processors, including the RP2040 that the Pico uses. While it is *in theory* possible to run the python implementation of Reticulum with some hacks on microcontrollers (via micropython), it is definitely not recommended ;) You are much better off using a compute device to handle routing, for now.
